osu.Server.Spectator/Hubs/Metadata/MetadataHub.cs

```diff
@@ -205,16 +212,36 @@ public async Task EndWatchingMultiplayerRoom(long id)
         protected override async Task CleanUpState(MetadataClientState state)
         {
             await base.CleanUpState(state);
-            await broadcastUserPresenceUpdate(state.UserId, null);
+            if (shouldBroadcastPresentToOtherUsers(state))
+                await broadcastUserPresenceUpdate(state.UserId, null);
             await scoreProcessedSubscriber.UnregisterFromAllMultiplayerRoomsAsync(state.UserId);
         }
 
         private Task broadcastUserPresenceUpdate(int userId, UserPresence? userPresence)
         {
-            if (userPresence?.Status == UserStatus.Offline)
-                userPresence = null;
+            // we never want appearing offline users to have their status broadcast to other clients.
+            Debug.Assert(userPresence?.Status != UserStatus.Offline);
 
             return Clients.Group(ONLINE_PRESENCE_WATCHERS_GROUP).UserPresenceUpdated(userId, userPresence);
         }
+
+        private bool shouldBroadcastPresentToOtherUsers(MetadataClientState state)
```

Comment:
Method name is typo'd:

```suggestion
        private bool shouldBroadcastPresenceToOtherUsers(MetadataClientState state)
```

(don't apply directly for obvious reasons)